### PR TITLE
Record-data shared component

### PR DESCRIPTION
**Pull request by lumburovskalina**
Implements:
- https://github.com/code-kern-ai/refinery/issues/196

### CODE REVIEW

File: src/app/base/components/record-card/record-card.component.ts
Comment (JWittmeyer) [inline src/app/base/components/record-card/record-card.component.ts]: this should be singular, right? we only display one record per card

File: src/app/base/components/record-card/record-card.component.ts
Comment (JWittmeyer) [inline src/app/base/components/record-card/record-card.component.ts]: I'd put the types into their own .ts file. Also, Record feels a bit too general imo. Since this is specifically needed for the databrowser and only works in the databrowser (or at least in combination with the extended record) I'd go with either ExtendedRecord or DatabrowserRecord as type name

File: src/app/base/components/record-card/record-card.component.ts
Comment (JWittmeyer) [reply src/app/base/components/record-card/record-card.component.ts]: ok i might have misread something here. Why is the extendedRecord needed? (the structure to hold all records in the record specific component)

File: src/app/base/components/record-card/record-card.component.ts
Comment (JWittmeyer) [inline src/app/base/components/record-card/record-card.component.ts]: why is that emitted? if this only stores it in the local storage so maybe no need to emit?

File: src/app/base/components/record-card/record-card.component.ts
Comment (JWittmeyer) [inline src/app/base/components/record-card/record-card.component.ts]: needed?

File: src/app/base/components/record-card/record-card.component.ts
Comment (JWittmeyer) [reply src/app/base/components/record-card/record-card.component.ts]: I see that the session ID is used somewhere in the HTML -> we can of course provide that. 


So in general a bit unclear on the direction we want to go with this component.
This could be A: To reuse the component (e.g. in the display for the "View" for run on 10 for heuristics) then we need to generalize a bit more (like record & attribute arrays and some optional stuff) 
Or B: Use this only as a code restructure tool only meant for the databrowser -> then we should organize it below the data browser in the structure of the folders


-> I think I would go with A to further reduce the complexity where we can -> but we can talk about this

File: src/app/base/components/record-card/record-card.component.ts
Comment (JWittmeyer) [inline src/app/base/components/record-card/record-card.component.ts]: the name suggests an array, the emitter is for a single number

File: src/app/base/components/record-card/record-card.component.ts
Comment (JWittmeyer) [inline src/app/base/components/record-card/record-card.component.ts]: if possible let's distance us from Maps a bit (I was the one introducing them however since they use functions (get) in the HTML this might not be optimal). Since we often use strings (uuids/attribute names) as keys they should be interchangeable. Objects keys etc. are more complex but often not necessary.

Also I'm not the biggest fan of the sortOrder/dict combination (Again something I introduced at some point I think) -> I think we should decide on a common approach for attributes throughout the UI.
A: double structure (dict+array+filter of array) -> benefit performance since only a small part is changed + dict access via key is fast
B: only array -> benefit redability -> will be replaced more often (maybe performance hit?) also maybe not since we need to loop anyway?

-> so maybe a global function to prepare an array? (again something we can talk about :) )

File: src/app/base/components/record-card/record-card.component.ts
Comment (JWittmeyer) [inline src/app/base/components/record-card/record-card.component.ts]: is the name of the function correct?

File: src/app/base/components/record-card/record-card.component.ts
Comment (lumburovskalina) [reply src/app/base/components/record-card/record-card.component.ts]: That's true, no need for emitting, moved the logic from the parent component

File: src/app/base/components/record-card/components/record-display/record-display.component.ts
Comment (JWittmeyer) [inline src/app/base/components/record-card/components/record-display/record-display.component.ts]: if the provided structure doesnt have an id we should throw an error here imo (so check if exists if yes then use it otherwise error)

File: src/app/base/components/record-card/components/record-display/record-display.component.ts
Comment (JWittmeyer) [inline src/app/base/components/record-card/components/record-display/record-display.component.ts]: shouldn't this be else if? or is this intentionally overwritten when both exist?

Comment (JWittmeyer) [reply src/app/base/components/record-card/components/record-display/record-display.component.ts]: also an else if neither exists with an error to ensure the developer uses it right

File: src/app/base/components/record-card/record-card-helper.ts
Comment (JWittmeyer) [inline src/app/base/components/record-card/record-card-helper.ts]: this doesn't make sense to me, why does the record card (singular instance) need a reference to the whole list?
As discussed before we should look for the stuff we explicitly need and only use that. For this instance I think an output event for clicked index should do it (the stage of the record ids would then be handled by the databrowser though)

File: src/app/base/components/record-card/components/record-display/record-display.component.ts
Comment (lumburovskalina) [reply src/app/base/components/record-card/components/record-display/record-display.component.ts]: Yes it should be else if (I think either recordData or fullRecordData can be replacement for data but not both of them at the same time)

File: src/app/base/components/record-card/record-card-helper.ts
Comment (JWittmeyer) [inline src/app/base/components/record-card/record-card-helper.ts]: i think just calling it recordIndex or currentRecordIndex would be more intuitive since the extendedRecord stuff is only loosely linked to the card component (since its located outside the folder scope)

**Comment by JWittmeyer** [comment]
Lets create a record-list component inside the data browser module

- [x] resolved

File: src/app/data/components/record-list/record-list.component.ts
Comment (JWittmeyer) [inline src/app/data/components/record-list/record-list.component.ts]: the name suggest to me that this is a number not an eventemmitter for a number. Maybe to generalize it a bit `recordClicked`?

File: src/app/data/components/record-list/record-list.component.ts
Comment (JWittmeyer) [inline src/app/data/components/record-list/record-list.component.ts]: maybe `similaritySearchRequested` ?

File: src/app/data/components/record-list/record-list.component.ts
Comment (JWittmeyer) [inline src/app/data/components/record-list/record-list.component.ts]: if this is fixed anyway why dont be use a constant reference which is initiialized from the get go?
something like this outside the class: 
```
const DATA_BROWSER_TABLE_COLUMN_HEADERS:ColumnData[] = [
  { field: 'type', displayName: 'Type', order: 1 },
  { field: 'task', displayName: 'Task', order: 2 },
  { field: 'label', displayName: 'Label', order: 3 },
  { field: 'amount', displayName: 'Amount', order: 4 },
  { field: 'confidenceAvg', displayName: 'Avg.confidence', order: 5 }
]
```

and then just reference it in the onInit.

With this we wouldn't need to recrate it every time the browser is opened

**Comment by JWittmeyer** [comment]
looks good. After the comments can be merged :)

**Review by JWittmeyer — APPROVED** [review]

**Pull request merged**
